cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -298,3 +298,50 @@ type AuditPolicyConfiguration struct {
 	LogMaxAge *int32
 	//TODO(chuckha) add other options for audit policy.
 }
+
+// CommonConfiguration defines the list of common configuration elements and the getter
+// methods that must exist for both the MasterConfiguration and NodeConfiguration objects.
+// This is used internally to deduplicate the kubeadm preflight checks.
+type CommonConfiguration interface {
```

Comment:
I haven't made up my mind yet whether this is a good or a bad abstraction at this level for the specific use case, but I can let this pass as a backport for the moment.